common/flatpak-run.c

```diff
@@ -2853,13 +2865,31 @@ flatpak_run_app (const char     *app_ref,
   gboolean generate_ld_so_conf = TRUE;
   gboolean use_ld_so_cache = TRUE;
   gboolean sandboxed = (flags & FLATPAK_RUN_FLAG_SANDBOX) != 0;
+  g_autoptr(EpcAppFilter) app_filter = NULL;
+  g_autoptr(GAsyncResult) app_filter_result = NULL;
 
   struct stat s;
 
   app_ref_parts = flatpak_decompose_ref (app_ref, error);
   if (app_ref_parts == NULL)
     return FALSE;
 
+  /* Check that this user is actually allowed to run this app. */
+  epc_get_app_filter_async (NULL, getuid (), TRUE, cancellable,
```

Comment:
Would it be so bad to add synchronous variants of these functions to the library? Every call-site I've seen so far (here and in `eos-parental-controls-client.py`) actually wants the function to be synchronous.

Reply:
I was thinking that too, but was rushing to get this finished. I think it would make a good follow-up.

Reply:
Thinking about it some more, it would add a significant amount of code to libeos-parental-controls, which couldn’t really be shared with the async implementation (since the library is essentially a chain of async calls, with no significant processing inbetween). I’m not sure that’s a maintenance burden worth taking on for the sake of avoiding some main context iterations in a client, given that I would expect the other clients to query/update the app filter synchronously.

I could be convinced otherwise, though, and would be happy to reassess this if the other client implementations turn out to need sync methods to call too.

Reply:
One way to share code would be to implement the async version in terms of the sync one, in a GTask thread.

Not a merge blocker, I don't think. Just a thought.